Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -252,6 +264,178 @@ namespace MiscCmds
 		return true;
 	}
 
+	void MiscCmds::LoadListOfReps()
+	{
+		INI_Reader ini;
+
+		string factionpropfile = "..\\data\\initialworld.ini";
+		if (ini.open(factionpropfile.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Group"))
+				{
+					uint ids_name;
+					string nickname;
+					while (ini.read_value())
+					{
+						if (ini.is_value("nickname"))
+						{
+							nickname = ini.get_value_string();
+						}
+						else if (ini.is_value("ids_name"))
+						{
+							ids_name = ini.get_value_int(0);
+						}
+
+					}
+					factions[nickname] = ids_name;
+				}
+			}
+			ini.close();
+			ConPrint(L"Rep: Loaded %u factions\n", factions.size());
+		}
+	}
+
+	string resetrepfileadress = "flhook_plugins\\playercntl_resetrep.ini";
+	map<wstring, uint> ShipnamesWhichAlreadyUsedResetRep;
+	void MiscCmds::Resetrep_Time_limits_load()
+	{
+		ShipnamesWhichAlreadyUsedResetRep.clear();
+		INI_Reader ini;
+		string factionpropfile = resetrepfileadress;
+		if (ini.open(factionpropfile.c_str(), false))
+		{
+			while (ini.read_header())
+			{
+				if (ini.is_header("Playerships"))
+				{
+					uint ids_name;
+					wstring nickname;
+					while (ini.read_value())
+					{
+						if (ini.is_value("shipname"))
+						{
+							ini_get_wstring(ini, nickname);
+						}
+						else if (ini.is_value("timelimit"))
+						{
+							ids_name = ini.get_value_int(0);
+							ShipnamesWhichAlreadyUsedResetRep[nickname] = ids_name;
```

Comment:
this should be done after the `while` block, it will otherwise break if someone tries to specify  `timelimit` before `shipname`

Reply:
fixed